### PR TITLE
gomplate: init at 3.6.0

### PR DESCRIPTION
**Pull request by jlesquembre**
###### Motivation for this change
Add [gomplate](https://docs.gomplate.ca)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/gomplate/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  preBuild = ''
+    export buildFlagsArray+=(
+      "-ldflags=
+        -w -s
+        -X ${goPackagePath}/v3/version.Version=${version}
+        -X ${goPackagePath}/v3/version.GitCommit=${rev}")
+  '';
+
+  goPackagePath = "github.com/hairyhenderson/gomplate";
```
Comment (doronbehar) [inline pkgs/development/tools/gomplate/default.nix:26]:
```suggestion
```

This is relevant only for `buildGoPackage`.

**Review by doronbehar — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/gomplate/default.nix (hunk 1 above)
Comment (jlesquembre) [reply pkgs/development/tools/gomplate/default.nix:26]: True, but I'm using it on the `buildFlagsArray`. I could change the variable name if that makes the intention clearer.

Comment (doronbehar) [reply pkgs/development/tools/gomplate/default.nix:26]: Then I guess it's not that important, keep it that way.

**Review by doronbehar — APPROVED** [review]

**Comment by 06kellyjac** [comment]
Since #99402 is merged this can be closed now

**Pull request closed without merging**